### PR TITLE
Enable secure boot

### PR DESCRIPTION
**Pull request by gaojiaqi7**
Enable secure boot for payload verification, according to the https://github.com/confidential-containers/td-shim/blob/e35f50087eb33e5220cb86f6c1bc36fa8637bc35/doc/secure_boot.md

### CODE REVIEW

**Comment by jyao1** [comment]
Please separate lto logger issue in another P-R.
It is not related to secure boot.

**Comment by jyao1** [comment]
Feedback:
1) please include sample secure boot key.
2) please move secure-boot-guide.md to doc dir. Only README.md is allowed in root dir.

File: secure-boot-guide.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+```
+
+### Sign payload
+Using ECDSA NIST P384 as example:
+```
+pushd rust-tdpayload-signing
+cargo run -- ../target/target/release/rust-td-payload 1 1 ECDSA_NIST_P384_SHA384 ../sample_key/ecdsa-p384-private.pk8
+popd
+```
+
```
Comment (jyao1) [inline secure-boot-guide.md:58]: Need explicit indicate where is the output file name and location.

Suggest to output final bin to same input dir, here it is "../target/target/release/"

File: secure-boot-guide.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+Enroll public key:
+```
+pushd rust-tdshim-key-enroll
+cargo run -- ../target/x86_64-unknown-uefi/release/final.bin ../sample_key/ecdsa-p384-public.der SHA384
+popd
+```
+
+The final.sb.bin with secure boot enabled is located in rust-tdshim-key-enroll.
```
Comment (jyao1) [inline secure-boot-guide.md:72]: Suggest to output final bin to same input dir, here it is "../target/x86_64-unknown-uefi/release/"

File: README.md [hunk 3]
```diff
@@ -46,6 +46,11 @@ set CC=clang
 set AR=llvm-ar
 ```
 
+### Secure boot support
+
+See [Secure Boot Guide](secure-boot-guide.md)
```
Comment (jyao1) [inline README.md:51]: Need move to doc dir.

Also need mention secure_boot.md just after "The API specification is at [td-shim specification]"

File: secure-boot-guide.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
+## Build td-shim
+
+```
+cargo xbuild -p rust-tdshim --features "secure-boot" --target x86_64-unknown-uefi --release
+```
+
+## Build payload
+
+Refer to [README.md], using ELF as example:
```
Comment (jyao1) [inline secure-boot-guide.md:9]: Need link?

File: secure-boot-guide.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
+## Build td-shim
```
Comment (jyao1) [inline secure-boot-guide.md:1]: Need one line to describe the high level feature.

And add link for [secure_boot.md].

Also, please add [secure_boot.md] to [td-shim spec.md]

File: secure-boot-guide.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        -pkeyopt rsa_keygen_pubexp:65537 | \
+        openssl pkcs8 -topk8 -nocrypt -outform der > rsa-3072-private.pk8
+```
+
+Generate public key file with private key:
+```
+openssl rsa -inform der -in rsa-3072-private.pk8 -pubout -outform der -out rsa-3072-public.der
+```
+
+### Sign payload
```
Comment (jyao1) [inline secure-boot-guide.md:51]: Add a link for rust-tdpayload-signing tool.

File: secure-boot-guide.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+Using ECDSA NIST P384 as example:
+```
+pushd rust-tdpayload-signing
+cargo run -- ../target/target/release/rust-td-payload 1 1 ECDSA_NIST_P384_SHA384 ../sample_key/ecdsa-p384-private.pk8
+popd
+```
+
+### Enroll public key
```
Comment (jyao1) [inline secure-boot-guide.md:59]: Add a link for rust-tdshim-key-enroll tool.

**Comment by gaojiaqi7** [comment]
> Feedback:
> 
> 1. please include sample secure boot key.
> 2. please move secure-boot-guide.md to doc dir. Only README.md is allowed in root dir.
> 3. please include a link for https://github.com/confidential-containers/td-shim/blob/staging/doc/secure_boot.md

All of these issues have been fixed.

**Pull request merged**